packages/eslint-plugin/tests/rules/no-useless-template-literals.test.ts

```diff
@@ -637,5 +691,79 @@ declare const nested: string, interpolation: string;
         },
       ],
     },
+
+    {
+      code: '`some ${/`/} string`;',
+      output: '`some /\\`/ string`;',
+      errors: [
+        {
+          messageId: 'noUselessTemplateLiteral',
+        },
+      ],
+    },
+    {
+      code: '`some ${/${}/} string`;',
+      output: '`some /\\${}/ string`;',
+      errors: [
+        {
+          messageId: 'noUselessTemplateLiteral',
+        },
+      ],
+    },
+    {
+      code: '`some ${/\\\\/} string`;',
+      output: '`some /\\\\\\\\/ string`;',
+      errors: [
+        {
+          messageId: 'noUselessTemplateLiteral',
+        },
+      ],
+    },
+
+    {
+      code: "`${'\\u00E5'}`;",
+      output: "'\\u00E5';",
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: "`${'\\n'}`;",
+      output: "'\\n';",
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: "` ${'\\u00E5'} `;",
+      output: '` \\u00E5 `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: "` ${'\\n'} `;",
+      output: '` \\n `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: noFormat`\` \${"\\n"} \`;`,
+      output: '` \\n `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: '` ${`\\n`} `;',
+      output: '` \\n `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: noFormat`\` \${ 'A\\u0307\\u0323' } \`;`,
+      output: '` A\\u0307\\u0323 `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: "` ${'👨‍👩‍👧‍👦'} `;",
+      output: '` 👨‍👩‍👧‍👦 `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
+    {
+      code: "` ${'\\ud83d\\udc68'} `;",
+      output: '` \\ud83d\\udc68 `;',
+      errors: [{ messageId: 'noUselessTemplateLiteral' }],
+    },
```

Comment:
I've thought of a few cruel test inputs that generate syntax errors on autofix. 

https://deploy-preview-8688--typescript-eslint.netlify.app/play#ts=5.4.2&fileType=.tsx&code=PTAEFMDcEsBtQK4DtwGcDGBDADuAJqHgPayyYBOoq0A5kqAPpJEAuDiq%2Bo09L4AttjJ8IAD2zk01IkgBQAAwAkigN4ByFQF81m%2BQG5ZskKAngAtGiy4CAI0zoA1i2iOFoVWoA68naH1A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUFIAEAuAngA4oDG0AlkXgLQrwUB2eA9IwPY2zKJLLI08iALZF4AQyE0GQ6OPioMkRNGjtokAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

Reply:
note - For the second case, i think i've seen in other contexts that a strategy for this is to only add an escape if the character needing escape is preceeded by an even number (possibly 0) of backslashes. less sure about the '$' shenanigans.

Reply:
What a nice catch! Thanks!